core/shared/utils/bh_log.c

```diff
@@ -51,7 +51,7 @@ bh_log(LogLevel log_level, const char *file, int line, const char *fmt, ...)
         os_printf("%s, line %d, ", file, line);
 
     va_start(ap, fmt);
-    os_vprintf(fmt, ap);
+    os_vprintf(log_level, fmt, ap);
```

Comment:
@eloparco this change introduces lots of code changes, and it makes the code structure a little strange, the platform layer seems to rely on the shared/util layer, and the prototype of os_vprintf is inconsistent with vprintf.
Per my understanding, you want to take control the log dump in your code, how about introducing another macro BH_VLOG like BH_VPRINTF, and call it when it is defined:
```C
    va_start(ap, fmt);
#if defined(BH_VLOG)
    BH_VLOG(log_level, fmt, ap);
#else
    os_vprintf(fmt, ap);
#endif
```
Or just defined BH_LOG to replace bh_log function:
```
#ifndef BH_LOG
void
bh_log(LogLevel log_level, const char *file, int line, const char *fmt, ...)
{
    ...
}
#endif
```

Reply:
Tried that just to make sure, but it doesn't allow to forward the wasm output, coming from `os_printf`, https://github.com/bytecodealliance/wasm-micro-runtime/blob/627debcefb1324a13d4f360a4346964fcde62dc5/core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c#L669 to the embedder (e.g. the output here https://github.com/bytecodealliance/wasm-micro-runtime/blob/627debcefb1324a13d4f360a4346964fcde62dc5/samples/basic/wasm-apps/testapp.c#L26 goes through `os_printf` but not `bh_log`).
What I mean is, overriding `bh_log` only allows to take control of the runtime output, not the wasm one.

Reply:
Can you define both BH_LOG and BH_VPRINTF? It will forward both bh_log and os_printf.